frontend/src/lib/StaticGraphParser.ts

```diff
@@ -15,196 +15,184 @@
+import { logger } from './Utils';
+
+export type nodeType = 'container' | 'dag' | 'unknown';
+
+export class SelectedNodeInfo {
+  public args: string[];
+  public command: string[];
+  public condition: string;
+  public image: string;
+  public inputs: string[][];
+  public nodeType: nodeType;
+  public outputs: string[][];
+
+  constructor() {
+    this.args = [];
+    this.command = [];
+    this.condition = '';
+    this.image = '';
+    this.inputs = [[]];
+    this.nodeType = 'unknown';
+    this.outputs = [[]];
+  }
 }
 
-export function createGraph(workflow: Workflow): dagre.graphlib.Graph {
-  const g = new dagre.graphlib.Graph();
-  g.setGraph({});
-  g.setDefaultEdgeLabel(() => ({}));
 
-  const NODE_WIDTH = 180;
-  const NODE_HEIGHT = 70;
+const NODE_WIDTH = 180;
+const NODE_HEIGHT = 70;
+
+export function _populateInfoFromTemplate(info: SelectedNodeInfo, template?: Template): SelectedNodeInfo {
+  if (!template || !template.container) {
+    return info;
+  }
+
+  info.nodeType = 'container';
+  info.args = template.container.args || [],
+  info.command = template.container.command || [],
+  info.image = template.container.image || '';
+
+  if (template.inputs) {
+    info.inputs =
+      (template.inputs.parameters || []).map(p => [p.name, p.value || '']);
+  }
+  if (template.outputs) {
+    info.outputs = (template.outputs.parameters || []).map(p => {
+      let value = '';
+      if (p.value) {
+        value = p.value;
+      } else if (p.valueFrom) {
+        value = p.valueFrom.jqFilter || p.valueFrom.jsonPath || p.valueFrom.parameter || p.valueFrom.path || '';
+      }
+      return [p.name, value];
+    });
+  }
+  return info;
+}
+
+/**
+ * Recursively construct the static graph of the Pipeline.
+ *
+ * @param graph The dagre graph object
+ * @param rootTemplateId The current template being explored at this level of recursion
+ * @param templates An unchanging map of template name to template and type
+ * @param parent The task that was being examined when the function recursed
+ */
+function buildDag(
+    graph: dagre.graphlib.Graph,
+    rootTemplateId: string,
+    templates: Map<string, { nodeType: nodeType, template: Template }>,
+    parent?: string): void {
+
+  const root = templates.get(rootTemplateId);
+
+  if (root && root.nodeType === 'dag') {
+    const template = root.template;
+
+    (template.dag.tasks || []).forEach((task) => {
+      // If the user specifies an exit handler, then the compiler will wrap the entire Pipeline
+      // within an additional DAG template prefixed with "exit-handler".
+      // If this is the case, we simply treat it as the root of the graph and work from there
+      if (task.name.startsWith('exit-handler')) {
+        buildDag(graph, task.template, templates);
+      } else {
+
+        // Parent here will be the task that pointed to this DAG template
+        if (parent) {
+          graph.setEdge(parent, task.name);
```

Comment:
Will this break if the task names are not globally unique?

Reply:
That depends on what you mean by break. It will not crash, but it may not display what you would expect? The nodes have only names as identifiers, so if there are two separate tasks with the same names, they will be represented by a single node within the graph, combining all of their inbound and outbound edges.

Reply:
Task names are only unique inside their dag namespace. It's OK to push this now, but iin future, the id of a task should be the full path `root-task/level1-task/level2-task/leaf-task` which won't clash with `root-task/level1-task/level2-2-task/leaf-task`

Reply:
I can do that in this PR pretty easily but with one caveat: for now, I'll have to assume that tasks can only have siblings as `dependencies`. Is this a safe assumption to make at this time?